src/components/cylc/workflow/Lumino.vue

```diff
@@ -15,187 +15,243 @@ You should have received a copy of the GNU General Public License
-      default: () => {}
-    },
-    workflowName: {
-      type: String,
-      required: true
-    },
-    /** All possible view component classes that can be rendered */
-    allViews: {
-      type: Array,
-      required: true
-    },
-  },
 
-  emits: [
-    'lumino:activated',
-    'lumino:deleted'
-  ],
+/**
+ * Mitt event for adding a view to the workspace.
+ * @typedef {Object} AddViewEvent
+ * @property {string} name - the view to add
+ * @property {Object=} initialOptions - prop passed to the view component
+ */
+
+const $eventBus = inject('eventBus')
+const $store = useStore()
 
-  beforeCreate () {
-    // Populate components
-    for (const { name, component } of this.allViews) {
-      this.$options.components[name] = component
-    }
+const props = defineProps({
+  workflowName: {
+    type: String,
+    required: true
   },
-
   /**
-   * Here we define the ID's for the Lumino DOM elements, and add the Dock panel to the main
-   * Box panel. In the next tick of Vue, the DOM element and the Vue element/ref are attached.
+   * All possible view component classes that can be rendered
+   *
+   * @type {Map<string, import('@/views/views.js').CylcView>}
    */
-  created () {
-    // create a box panel, which holds the dock panel, and controls its layout
-    this.box = new BoxPanel({ direction: 'left-to-right', spacing: 0 })
-    // create dock panel, which holds the widgets
-    this.dock = new DockPanel()
-    // Note: box & dock must not be in data() as the functionality breaks if
-    // the lumino objects are proxied by Vue
-
-    this.box.addWidget(this.dock)
-    BoxPanel.setStretch(this.dock, 1)
-
-    const resizeObserver = new ResizeObserver(() => {
-      this.box.update()
-    })
-
-    this.$nextTick(() => {
-      // Attach box panel to DOM:
-      Widget.attach(this.box, this.$refs.main)
-      // Watch for resize of the main element to trigger relayout:
-      resizeObserver.observe(this.$refs.main)
-    })
+  allViews: {
+    type: Map,
+    required: true
   },
+})
 
-  computed: {
-    /**
-     * We want to watch this.views; however, the (newVal, oldVal) args
-     * do not differ when a deeply watched object's properties change.
-     * So here is a workaround.
-     */
-    _views () {
-      return Object.assign({}, this.views)
-    }
-  },
+const emit = defineEmits([
+  'emptied'
+])
+
+/**
+ * Template ref
+ * @type {import('vue').Ref<HTMLElement>}
+ */
+const mainDiv = ref(null)
+
+/**
+ * Mapping of widget ID to the name of view component and its initialOptions prop.
+ *
+ * @type {import('vue').Ref<Map<string, AddViewEvent>>}
+ */
+const views = ref(new Map())
+
+const defaultView = useDefaultView()
+
+// create a box panel, which holds the dock panel, and controls its layout
+const boxPanel = new BoxPanel({ direction: 'left-to-right', spacing: 0 })
+// create dock panel, which holds the widgets
+const dockPanel = new DockPanel()
+boxPanel.addWidget(dockPanel)
+BoxPanel.setStretch(dockPanel, 1)
```

Comment:
Is that meant to be `BoxPanel.setStretch` with a capital `B`?

Reply:
Yes, same as before